From: Ben Hutchings <ben@decadent.org.uk>
Subject: Revert "VM: add vm_munmap() helper function" in i810
Date: Sun, 02 Dec 2012 23:49:49 +0000

This reverts commit a46ef99d80817a167477ed1c8b4d90ee0c2e726f 'VM: add
"vm_munmap()" helper function' and commit
bfce281c287a427d0841fadf5d59242757b4e620 'kill mm argument of
vm_munmap()' in drivers/gpu/drm/i810/i810_dma.c

--- a/drivers/gpu/drm/i810/i810_dma.c
+++ b/drivers/gpu/drm/i810/i810_dma.c
@@ -157,8 +157,11 @@
 	if (buf_priv->currently_mapped != I810_BUF_MAPPED)
 		return -EINVAL;
 
-	retcode = vm_munmap((unsigned long)buf_priv->virtual,
+	down_write(&current->mm->mmap_sem);
+	retcode = do_munmap(current->mm,
+			    (unsigned long)buf_priv->virtual,
 			    (size_t) buf->total);
+	up_write(&current->mm->mmap_sem);
 
 	buf_priv->currently_mapped = I810_BUF_UNMAPPED;
 	buf_priv->virtual = NULL;
